android/src/com/mapswithme/maps/routing/RoutingController.java

```diff
@@ -648,20 +638,63 @@ public boolean cancel()
       mLogger.d(TAG, "cancel: navigating");
 
       cancelInternal();
+      cancelNavigation();
       if (mContainer != null)
       {
-        mContainer.showNavigation(false);
         mContainer.updateMenu();
       }
-      if (mContainer != null)
-        mContainer.onNavigationCancelled();
       return true;
     }
 
     mLogger.d(TAG, "cancel: none");
     return false;
   }
 
+  public void startPlanning()
+  {
+    if (mContainer != null)
+    {
+      showRoutePlan();
+      mContainer.onPlanningStarted();
+    }
+  }
+
+  public void startPlanning(final @Nullable MapObject startPoint, final @Nullable MapObject endPoint)
+  {
+    if (mContainer != null)
```

Comment:
Why is this check really needed in all these functions? When this variable can be null?

I'm asking because I know that in the past often developers did not investigate (and fix) the true cause, but did a "hotfix-workaround" instead.

Reply:
I kept those checks because they were present before and I did not want to introduce new bugs/crashes.

From what I could see, `mContainer` is set in the `attach` method, called in the `MwmActivity` `onStart` lifecycle event. and deleted in the `onStop`.

As this field can be null, it is annotated with `@Nullable` thus Android Studio will complain when omitting this check.

Reply:
Does it mean that these functions may be called when the main activity is stopped? If yes, what are these cases?

Reply:
I don't think those functions can be called when the main activity is stopped, but as the field is nullable not having the check creates warnings.

Properly removing those checks without creating warnings would mean a bigger refactoring, probably to be done in an other PR.